Userland/Utilities/nc.cpp

```diff
@@ -42,18 +45,48 @@ int main(int argc, char** argv)
     bool should_listen = false;
     bool verbose = false;
     bool should_close = false;
+    bool udp_mode = false;
     const char* addr = nullptr;
     int port = 0;
 
     Core::ArgsParser args_parser;
     args_parser.set_general_help("Network cat: Connect to network sockets as if it were a file.");
     args_parser.add_option(should_listen, "Listen instead of connecting", "listen", 'l');
     args_parser.add_option(verbose, "Log everything that's happening", "verbose", 'v');
+    args_parser.add_option(udp_mode, "UDP mode", "udp", 'u');
     args_parser.add_option(should_close, "Close connection after reading stdin to the end", nullptr, 'N');
     args_parser.add_positional_argument(addr, "Address to connect to or listen on", "address");
     args_parser.add_positional_argument(port, "Port to connect to or listen on", "port");
     args_parser.parse(argc, argv);
 
+    if (udp_mode) {
+        if (should_listen) {
+            fprintf(stderr, "listening on UDP not yet supported\n");
+            exit(1);
+        }
+
+        Core::EventLoop loop;
+        auto socket = Core::UDPSocket::construct();
+
+        socket->on_connected = [&]() {
+            if (verbose)
+                fprintf(stderr, "connected to %s:%d\n", addr, port);
```

Comment:
```suggestion
                warnln("connected to {}:{}", addr, port);
```

you did the right thing since we already log to stderr below.... but any idea why? Can't this just go to stdout?

if you'd agree with that and maybe want to fix up the other calls as well:

```suggestion
                outln("connected to {}:{}", addr, port);
```

Reply:
I've personally used nc in the form of `nc -l someport > out.file` to transfer files over the network, so that would interfere with that, but im not sure if thats intended functionality

Reply:
hmm, that's a good point - let's keep it this way then.

Reply:
I'll add a comment to clarify since it is a bit confusing